src/sql-parser/src/parser.rs

```diff
@@ -55,6 +57,14 @@ macro_rules! parser_err {
     };
 }
 
+macro_rules! parser_statement_err {
+    ($parser:expr, $statement: expr, $pos:expr, $MSG:expr) => {
+        Err($parser.statement_error($statement, $pos, $MSG.to_string()))
+    };
+    ($parser:expr, $statement: expr, $pos:expr, $($arg:tt)*) => {
+        Err($parser.statement_error($statement, $pos, format!($($arg)*)))
+    };
+}
```

Comment:
This macro is called only twice. I think it'd be better to just do `parser_err!().map_no_statement_err()` (like you do in a few places already).